lib/auth/saml.go

```diff
@@ -148,21 +178,53 @@ func (a *Server) getSAMLProvider(conn types.SAMLConnector) (*saml2.SAMLServicePr
 	return serviceProvider, nil
 }
 
-func (a *Server) calculateSAMLUser(connector types.SAMLConnector, assertionInfo saml2.AssertionInfo, request *services.SAMLAuthRequest) (*createUserParams, error) {
-	var err error
+type ssoDiagContext struct {
```

Comment:
Since this is probably going to be reused b/w different connector types, I think it's worth moving it into its own file.